network/nordvpn/doinst.sh

```diff
@@ -1,3 +1,14 @@
+# Patched by Peter Christy <pete.christy.uk@gmail.com>
+
+# update configuration and shared library cache for linker to find .so files
+echo "/usr/$(uname -m | grep -q 'x86_64' && echo lib64 || echo lib)/nordvpn" > /etc/ld.so.conf.d/nordvpn.conf
```

Comment:
I don't think we want to add these to the whole system. Probably better to just make a wrapper that sets LD_CONFIG only for nordvpn?

Reply:
I was thinking the same about those libs, but there is no way to set the PATH to /usr/lib64/nordvpn without a bash script or is there something else? so the user have to do this manually? 
I want to go for LD_CONFIG and just set PATH in a bash script in my pkg

Reply:
I just did a rewrite of the init script but it will take some time to test this. there could be bugs but IA dont find it
```
#!/usr/bin/env bash

# Short-Description: Protect your privacy online and access media content with no regional restrictions.
# Strong encryption and no-log policy with 3000+ servers in 60+ countries.
### END INIT INFO

# Reset PATH to a safe and sane default paths
# /usr/lib64/nordvpn have both binary and lib files
export PATH=/usr/lib64/nordvpn:/sbin:/bin:/usr/sbin:/usr/bin
export LD_LIBRARY_PATH=/usr/lib64/nordvpn:/lib64/:/usr/lib64

BASE=nordvpn
DAEMON="/usr/sbin/${BASE}d"
PID=/var/run/$BASE.pid
SOCK=/run/$BASE/${BASE}d.sock
SOCKDIR=/run/$BASE

#
# Function to let nordvpn users run the nordvpn command
#
nordvpnd_give_socket() {
  chown root:$BASE $SOCK
}

#
# Function to take socket ownership back to only root user
#
nordvpnd_take_socket() {
  chown root:root $SOCK
}

#
# Function to setup the socket for the users
#
create_socket_dir() {
  if [[ -d $SOCKDIR ]]; then
    return
  fi
  mkdir -m 0770 $SOCKDIR
  nordvpnd_give_socket
}

#
# Function to ensure the script is run as root
#
check_root() {
  if [ "$(id -u)" -ne 0 ]; then
    echo "This script must be run as root."
    exit 1
  fi
}

#
# Function to check for existing instances of the daemon
#
check_running_instance() {
  if pidof nordvpnd >/dev/null 2>&1; then
    echo "NordVPN daemon is already running."
    exit 1
  fi
}

#
# Function that starts the daemon
#
nordvpnd_start()
{
  # this might be to much, no other user should run this ever
  # and never 2 processes with different users
  check_root
  check_running_instance
  create_socket_dir
  rm -rf $SOCK
  if [ -s $PID ]; then
    echo "$BASE is already running: $(cat $PID)"
    exit 1
  fi

  if [ -x $DAEMON ]; then
    echo "Starting $BASE"
    sleep 1
    $DAEMON > /dev/null & # we can maybe use daemon logging from here
    pidof $DAEMON > $PID
    sleep 5 # this was set on 10 before, that is to long? This is a test for now
    if [ -S "$SOCK" ]; then
      echo "$BASE has been started."
    else
      echo "$BASE failed to start. Please restart the daemon."
    fi
  fi
}

#
# Function that stops the daemon
#
nordvpnd_stop()
{
  if [ -s $PID ]; then
    echo "Stopping $BASE"
    sleep 1
    kill "$(cat "$PID")"
    rm -rf $PID $SOCK
    sleep 1
    echo "$BASE has been stopped."
  else
    echo "$BASE is not running."
  fi
}

#
# Function that restarts the daemon
#
nordvpnd_restart()
{
  nordvpnd_stop
  sleep 1
  nordvpnd_start
}

#
# Function that shows the current status of the daemon
#
nordvpnd_status()
{
  if [ -s $PID ]; then
    echo "$BASE is running: $(cat $PID)"
  else
    echo "$BASE is not running."
  fi
}


case "$1" in
  start)
    nordvpnd_start
    ;;
  stop)
    nordvpnd_stop
    ;;
  restart)
    nordvpnd_restart
    ;;
  status)
    nordvpnd_status
    ;;
  give_socket)
    nordvpnd_give_socket
    ;;
  take_socket)
    nordvpnd_take_socket
    ;;
  *)
    echo $"Usage: $0 {start|stop|restart|status|give_socket|take_socket}"
    exit 1
    ;;
esac

```

Reply:
yes this works and fix it all + meshnet
just put nordvpn binary in /usr/lib64/nordvpn

```
#!/bin/bash
export PATH=/usr/lib64/nordvpn:/sbin:/bin:/usr/sbin:/usr/bin
export LD_LIBRARY_PATH=/usr/lib64/nordvpn:/lib64/:/usr/lib64
exec /usr/lib64/nordvpn/nordvpn "$@"
```
have fun, Im out :)

Reply:
you shouldn't put executables in /usr/lib{64}
that's for library

Reply:
yes, or /usr/bin

Reply:
I have tested this for some time back and forth with all possible settings. I have no disconnects and no problems. You need to take some action.
```
cd "$PKG"
# Libs for 64bit only here
mkdir usr/lib64
mv usr/lib/nordvpn usr/lib64/
rm -fr etc usr/lib # What's left is unneeded for Slackware
chmod 755 usr/lib64/nordvpn/libsqlite3.so
chmod 755 usr/lib64/nordvpn/libtelio.so

# Helper programs not useful to run outside the app
mkdir -p usr/libexec/nordvpn
mv usr/lib64/nordvpn/nordfileshare usr/libexec/nordvpn/
mv usr/lib64/nordvpn/norduserd usr/libexec/nordvpn/
mv usr/lib64/nordvpn/openvpn usr/libexec/nordvpn/

# Used with wrapper
mv usr/bin/nordvpn usr/libexec/nordvpn/
```
```
#!/bin/bash
export PATH=/usr/libexec/nordvpn:/sbin:/bin:/usr/sbin:/usr/bin
export LD_LIBRARY_PATH=/usr/lib64/nordvpn:/lib64/:/usr/lib64
exec /usr/libexec/nordvpn/nordvpn "$@"

```

```
#!/usr/bin/env bash

# Short-Description: Protect your privacy online and access media content with no regional restrictions.
# Strong encryption and no-log policy with 3000+ servers in 60+ countries.
### END INIT INFO

# Reset PATH to a safe and sane default paths
export PATH=/usr/libexec/nordvpn:/sbin:/bin:/usr/sbin:/usr/bin
export LD_LIBRARY_PATH=/usr/lib64/nordvpn:/lib64/:/usr/lib64

BASE=nordvpn
DAEMON="/usr/sbin/${BASE}d"
PID=/var/run/$BASE.pid
LOGFILE=/var/log/nordvpn.log
SOCK=/run/$BASE/${BASE}d.sock
SOCKDIR=/run/$BASE

#
# Function to setup the socket for the users
#
create_socket_dir() {
  if [[ -d $SOCKDIR ]]; then
    return
  fi
  mkdir -m 0770 $SOCKDIR
}

#
# Function to ensure the script is run as root
#
check_root() {
  if [ "$(id -u)" -ne 0 ]; then
    echo "This script must be run as root."
    exit 1
  fi
}

#
# Function to check for existing instances of the daemon
#
check_running_instance() {
  if pidof nordvpnd >/dev/null 2>&1; then
    echo "NordVPN daemon is already running."
    exit 1
  fi
}

#
# Function that starts the daemon
#
nordvpnd_start()
{
  check_root
  check_running_instance
  create_socket_dir
  chown root:$BASE $SOCKDIR
  rm -rf $SOCK
  if [ -s $PID ]; then
    echo "$BASE is already running: $(cat $PID)"
    exit 1
  fi

  if [ -x $DAEMON ]; then
    echo "Starting $BASE"
    if [ "$1" = "LOG" ]; then
      touch $LOGFILE
      chmod 640 $LOGFILE
      $DAEMON >> $LOGFILE 2>&1 &
    else
      $DAEMON > /dev/null 2>&1 &
    fi
    pidof $DAEMON > $PID

    for i in {1..10}; do
      if [ -S "$SOCK" ]; then
        chown root:$BASE $SOCK
        echo "$BASE has been started."
        break
      fi
      sleep 1
    done

    if [ ! -S "$SOCK" ]; then
      echo "$BASE failed to start the daemon."
    fi
  fi
}

#
# Function that stops the daemon
#
nordvpnd_stop()
{
  check_root
  if [ -s $PID ]; then
    echo "Stopping $BASE"
    sleep 1
    kill "$(cat "$PID")"
    rm -rf $PID $SOCK
    sleep 1
    echo "$BASE has been stopped."
  else
    echo "$BASE is not running."
  fi
}

#
# Function that restarts the daemon
#
nordvpnd_restart()
{
  nordvpnd_stop
  if [ -s $PID ]; then
    echo "Waiting for $BASE to stop..."
    while kill -0 "$(cat "$PID")" 2>/dev/null; do
      sleep 1
    done
  fi
  echo "$BASE has stopped. Restarting now."
  nordvpnd_start
}

#
# Function that shows the current status of the daemon
#
nordvpnd_status()
{
  if [ -s $PID ]; then
    echo "$BASE is running: $(cat $PID)"
  else
    echo "$BASE is not running."
  fi
}


case "$1" in
  start)
    nordvpnd_start
    ;;
  stop)
    nordvpnd_stop
    ;;
  restart)
    nordvpnd_restart
    ;;
  status)
    nordvpnd_status
    ;;
  start_with_logging)
    nordvpnd_start "LOG"
    ;;
  *)
    echo $"Usage: $0 {start|stop|restart|status|start_with_logging}"
    exit 1
    ;;
```

Reply:
it's up to the maintainer